### ISSUE TITLE
Local backup import issues on Heroku

### ISSUE THREAD

**Issue by kairosart** [juice-shop/juice-shop#1579]
I've deployed Juice Shop on my Heroku account from @bkimminich github page and everything seems to be right. I save my session on my disk and the next day I try to continue with my work and there are only some goals left but not all I've done. I try to upload the saved file but nothing happens. Even the user I've created has disappeared.

**Comment by bkimminich** [comment]
The user deletion is expected behavior, as the database is cleared on any restart, and free Heroku dynos restart all the time as they're put to sleep and wake up again.

The progress export issues I'll look into.

**Comment by bkimminich** [comment]
Okay, here's what I tested: I exported the progress from https://demo.owasp-juice.shop. Then I started Juice Shop _locally_ and opened it in an incognito window. I dismissed all banners, set some random language, and then went to the Score Board (which solved the associated challenge). There I imported the backup from the demo instance. The effect was:

1. The language was changed immediately
2. The challenge progress showed only the Score Board challenge solved as before

I restarted my local server while remaining on the Score Board. After the restart the imported continue code cookie was seen and sent to the server where all the challenges were solved. This triggered back the websocket call to my browser where now all the challenges were marked solved on the Score Board and I got all the green notifications, too.

So, the process itself works, but **the snackbar message after the import should say that a restart is needed to apply challenge progress changes**. I'll add that, because it's not obvious, but there's no way around it due to the way progress is handled with the websockets etc.

Next I went to https://kairosjuiceshop.herokuapp.com/#/score-board and imported the backup file from demo.owasp-juice.shop there. Language changes are applied as expected, continue code is set, too. But as I can't restart it, the progress isn't applied. I tried the same on a second Heroku instance I do have under control (http://preview.owasp-juice.shop/#/score-board) and it behaves exactly like my localhost instance as described above. I need to do the restart there, too.

Here's what I _suspect_ is going wrong when you use the backup restore: You have some browser tab still open your instance where the continue code from the backup is not set. If that tab is the fastest to notice the server start, it will send back its continue code and you get not the expected challenges solved. To verify this, here's what you can test:

1. Close all your browsers.
2. Open a single incognito browser tab on https://kairosjuiceshop.herokuapp.com/#/score-board
3. Set some language other than English
4. Set some weird Score Board filters, like turning off every second vuln category
5. Import the JSON backup attached here (it has a whole lot of challenges solved)
6. Hit F5
6. You should now (or even before F5 key) immediately see the language and filters change
7. The Score Board progress _will not change_
8. Keep that https://kairosjuiceshop.herokuapp.com/#/score-board tab open and don't do anything there
8. In another tab/windows go to your Heroku dashboard (should be https://dashboard.heroku.com/apps/kairosjuiceshop)
9. Click "More" and "Restart all dynos"
10. Wait a short while and watch the https://kairosjuiceshop.herokuapp.com/#/score-board tab. It should trigger a ton of challenge notifications. If nothing happens for a minute, then F5 and see if now the notifications arrive. It should look like this:

![image](https://user-images.githubusercontent.com/3531020/106531312-48204300-64ee-11eb-8960-4d07b3f9cd81.png)

TL;DR: I'll improve the **documentation** mentioning the **necessary server restart**. The websocket race condition thing from multiple browsers, I'll put into a **troubleshooting** tip in the ebook.

**Comment by bkimminich** [comment]
Forget all the above. I managed to trigger the progress restore on the server without even having to restart the application. All you now need to do is click _Apply changes now_ in the snackbar and the progress and all other settings will be applied immediately.

![image](https://user-images.githubusercontent.com/3531020/106532771-1b215f80-64f1-11eb-8719-8137aa375636.png)

The snackbar shows 10sec instead of 5sec, and if you don't click that action, then all my text above about server restart and race conditions between browser windows etc.etc. apply again.

**Comment by bkimminich** [comment]
This should do as documentation, I hope...

![image](https://user-images.githubusercontent.com/3531020/106534265-04303c80-64f4-11eb-9da1-f4e33996c212.png)
